src/Agent/NewRelic/Agent/Core/Samplers/GCSamplerV2ReflectionHelper.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Linq.Expressions;
+using System.Reflection;
+using NewRelic.Agent.Extensions.Logging;
+using NewRelic.Reflection;
+
+namespace NewRelic.Agent.Core.Samplers
+{
+    // to allow for unit testing
+    public interface IGCSamplerV2ReflectionHelper
+    {
+        Func<object, object> GetGenerationInfo { get; }
+        bool ReflectionFailed { get; }
+        Func<object, object> GCGetMemoryInfo_Invoker { get; }
+        Func<object, object> GCGetTotalAllocatedBytes_Invoker { get; }
+    }
+
+    public class GCSamplerV2ReflectionHelper : IGCSamplerV2ReflectionHelper
+    {
+        public Func<object, object> GetGenerationInfo { get; private set; }
+        public bool ReflectionFailed { get; private set; }
+        public Func<object, object> GCGetMemoryInfo_Invoker { get; private set; }
+        public Func<object, object> GCGetTotalAllocatedBytes_Invoker { get; private set; }
+
+        public GCSamplerV2ReflectionHelper()
+        {
+            try
+            {
+                var assembly = Assembly.Load("System.Runtime");
+                var gcType = assembly.GetType("System.GC");
+                var paramType = assembly.GetType("System.GCKind");
+                var returnType = assembly.GetType("System.GCMemoryInfo");
+
+                if (!VisibilityBypasser.Instance.TryGenerateOneParameterStaticMethodCaller(gcType, "GetGCMemoryInfo", paramType, returnType, out var accessor))
+                {
+                    ReflectionFailed = true;
+                }
+                else
+                    GCGetMemoryInfo_Invoker = accessor;
+
+                if (!ReflectionFailed)
+                {
+                    paramType = assembly.GetType("System.Boolean");
+                    returnType = assembly.GetType("System.Int64");
```

Comment:
I think this can be simplified to typeof(bool) and typeof(long), which more closely matches the actual signature.

Reply:
Good callout. Making the change now.